beacon-chain/sync/validate_attester_slashing.go

```diff
@@ -33,10 +34,11 @@ func (r *RegularSync) validateAttesterSlashing(ctx context.Context, msg proto.Me
 		return false, nil
 	}
 
-	slashing, ok := msg.(*ethpb.AttesterSlashing)
```

Comment:
Wait, why are you undoing this? This is the appropriate way to safely do type assertions.

Reply:
https://github.com/prysmaticlabs/prysm/pull/4164#discussion_r352691884

from @nisdas

Reply:
I added type assertions for all earlier but he mentioned previous step in the pipeline verified this

Reply:
ok, i don't feel great about this since it's safer to handle it the other way, even if something else is checking it (where?). 

Defensive programming 101: I am my own biggest enemy.

Reply:
Yeah. I agree with you guys. Will revert back to the original